src/dex/usdc-transmuter/usdc-transmuter.ts

```diff
@@ -0,0 +1,185 @@
+    const destTokenAddress = destToken.address.toLowerCase();
+
+    return (
+      (this.isUSDC(srcTokenAddress) && this.isUSDCe(destTokenAddress)) ||
+      (this.isUSDCe(srcTokenAddress) && this.isUSDC(destTokenAddress))
+    );
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return null;
+  }
+
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    if (this.isAppropriatePair(srcToken, destToken)) {
+      return [`${this.dexKey}_${srcToken.address}_${destToken.address}`];
+    }
+
+    return [];
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<UsdcTransmuterData>> {
+    if (!this.isAppropriatePair(srcToken, destToken)) {
+      return null;
+    }
+
+    return [
+      {
+        prices: [...amounts],
+        unit: this.unitPrice,
+        data: null,
+        exchange: this.dexKey,
+        gasCost: USDC_TRANSMUTER_GAS_COST,
+        poolAddresses: [this.config.usdcTransmuterAddress],
+      },
+    ];
+  }
+
+  getCalldataGasCost(
+    poolPrices: PoolPrices<UsdcTransmuterData> | null,
+  ): number {
+    return CALLDATA_GAS_COST.FUNCTION_SELECTOR + CALLDATA_GAS_COST.AMOUNT;
+  }
+
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: UsdcTransmuterData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    return {
+      targetExchange: this.config.usdcTransmuterAddress,
+      payload: '0x',
+      networkFee: '0',
+    };
+  }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: UsdcTransmuterData,
+    side: SwapSide,
+  ): DexExchangeParam {
+    const swapData = this.usdcTransmuterIface.encodeFunctionData(
+      this.isUSDC(srcToken)
+        ? UsdcTransmuterFunctions.deposit
+        : UsdcTransmuterFunctions.withdraw,
+      [srcAmount],
+    );
+
+    return {
+      needWrapNative: this.needWrapNative,
+      dexFuncHasRecipient: false,
+      exchangeData: swapData,
+      targetExchange: this.config.usdcTransmuterAddress,
+      returnAmountPos: undefined,
+    };
+  }
+
+  async getTopPoolsForToken(
+    tokenAddress: Address,
+    limit: number,
+  ): Promise<PoolLiquidity[]> {
+    if (this.isUSDC(tokenAddress) || this.isUSDCe(tokenAddress)) {
+      const isUSDC = this.isUSDC(tokenAddress);
+
+      return [
+        {
+          address: this.config.usdcTransmuterAddress,
+          connectorTokens: [
+            isUSDC ? this.config.usdceToken : this.config.usdcToken,
+          ],
+          exchange: this.dexKey,
+          liquidityUSD: 1000000000, // Just returning a big number so this DEX will be preferred
```

Comment:
think we can use the new `UNLIMITED_USD_LIQUIDITY` here, but need @KanievskyiDanylo to verify